### ISSUE TITLE
The welcome guide isn't translated on atomic sites when the Gutenberg plugin is deactivated

### ISSUE THREAD

**Issue by p-jackson** [Automattic/wp-calypso#53364]
### Steps to reproduce the behavior

1. Go to an atomic site
2. Deactivate the Gutenberg plugin
3. Switch the language to non-`en`
4. Open editor and show the welcome guide

#### What I expected to happen

The welcome guide should be translated

#### What actually happened
<img width="1275" alt="Screenshot 2021-06-03 at 11 45 54 AM" src="https://user-images.githubusercontent.com/1500769/120607749-98abab00-c4a4-11eb-9e86-c4c0a455df64.png">

### Context
I originally tried to translate the welcome tour in #51567. Unfortunately this depended on the latest version of GB being available. Since ETK needs to work with the GB plugin disabled this caused a crash. The crash was fixed by #53335, but we still need to go back and fix the translation problem. Probably by wrapping the welcome guide UI in a `<LocaleProvider>` like other places in ETK do.

**Comment by roo2** [comment]
hmmm we are already surrounding the welcome tour with the `<LocaleProvider>` [source](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/block-editor-nux.js#L63) and it is getting the correct locale. The call to `useLocale()` returns the correct locale without relying on `i18n.getLocaleData` [source](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-launch.js#L61).

I'll continue investigating another solution to why this content isn't being translated when gutenberg is disabled

**Comment by roo2** [comment]
it turns out that `getLocaleData` is called (and thus, was crashing before the [fix](https://github.com/Automattic/wp-calypso/pull/53335)) even if the LocaleProvider is set up correctly.

I believe that somewhere behind the scenes in the i18n framework (which I currently don't understand very deeply) we are relying on `getLocaleData` or some other dependency or function that only exists when gutenberg is installed.

The root cause of this is that the dependency extraction plugin [doesn't keep track of minimum required versions](https://github.com/Automattic/wp-calypso/issues/53306#issuecomment-852172606)

**Comment by p-jackson** [comment]
That's weird, if `<LocaleProvider>` is already there and getting the correct locale, then I wonder how I was able to get the screenshot in the description above 🤔

**Comment by roo2** [comment]
yea, I originally added the LocaleProvider so that the links at the end of the tour would be localized in https://github.com/Automattic/wp-calypso/pull/52099, [via useLocale](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-launch.js#L61). This part _is_ working without gutenberg installed, but the actual translation doesn't seem to be using the `<LocaleProvider>`

In this screenshot, the link is correctly localized to `/es/` but the content is not translated into Spanish.
<img width="801" alt="Screen Shot 2021-06-17 at 3 43 05 PM" src="https://user-images.githubusercontent.com/22446385/122327984-eafbca00-cf82-11eb-9446-0c3bb159352e.png">

**Comment by p-jackson** [comment]
Double checked that the translations exist: https://translate.wordpress.org/projects/wp-plugins/full-site-editing/stable/es/default/?filters[term]=your+site+is+private+until&filters[term_scope]=scope_any&filters[status]=current_or_waiting_or_fuzzy_or_untranslated&filters[user_login]=&filter=Apply+Filters&sort[by]=priority&sort[how]=desc

They do, so that's not the problem

**Comment by roo2** [comment]
Interestingly, I first reproduced this on a (jurassic ninja) site which never had the Gutenberg plugin installed. After I installed Gutenberg on the site, the welcome tour was translated, as expected.  Surprisingly, after I uninstalled Gutenberg, the welcome tour _remained_ translated.

**Comment by roo2** [comment]
I couldn't reproduce this either with my atomic platform site where I deactivated gutenberg, only on a site where gutenberg has never been installed.

**Comment by roo2** [comment]
Trying to reproduce this I found some very interesting caching behaviour:

When changing languages, the new language will not be translated (english will be shown) unless it has already been used.
If a new version of the ETK is uploaded, (add a whitespace character and rebuild), then the new language will be translated.
If you change back to a language that you have previously had working, the language will work as expected.

Interestingly, this happened **even after I re-activated gutenberg** on my atomic test site.

There seems to be some kind of language caching happening in the ETK which prevents new language translations from being used. This cache is cleared when I install a new version of the ETK, allowing the new language translations to be used.

This seems to explain some strange behaviour that @p-jackson reported p1623969049287100-slack-C01VA0R5NKY

I tested (with a log statement) that the `<LocaleProvider` [wrapping the welcome tour](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/block-editor-nux.js#L63) was getting the correct locale passed to it in all cases.
